components/proxy-service/internal/metadata/remoteenv/util.go

```diff
@@ -41,3 +40,16 @@ func convertFromK8sType(service v1alpha1.Service) (Service, apperrors.AppError)
 		Events:              events,
 	}, nil
 }
+
+func convertCredentialsFromK8sType(credentials v1alpha1.Credentials) *Credentials {
```

Comment:
It would be a good idea to have both of these methods unit tested.

Reply:
Actually I'm not convinced if adding this test is vital. Those functions are tested by repository_test.Adding a separate test for functions defined in utils.go won't make repository_test.go any shorter. Maybe we should move the contents of util.go to repository.go as they are detail of repository?